src/App.tsx

```diff
@@ -1,26 +1,151 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodo, getTodos, postTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterTypes';
+import { filterTodos } from './utils/helpers';
+import { ErrorTypes } from './types/ErrorTypes';
+import { Header } from './Components/Header';
+import { Footer } from './Components/Footer';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { TodoList } from './Components/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [error, setError] = useState<string>('');
+  const [value, setValue] = useState('');
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [isLoading, setIsLoading] = useState<number>(0);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, error]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError(ErrorTypes.OneMessage);
+        setTimeout(() => {
+          setError('');
+        }, 3000);
+      });
+  }, []);
+
+  const removeTodo = (todoId: number) => {
+    setIsLoading(todoId);
```

Comment:
Instead of setting a single id you can create a state with an array of loading todo ids. So you can add and remove ids of todos that currently loading and show loaders for them. It can help with the next task